install-scripts/mac/skm-install-mac.sh

```diff
@@ -0,0 +1,128 @@
+
+HOME_PATH=~
+INSTALL_PATH="${HOME_PATH}/.splashkit"
+
+function has_git() {
+	git --help 2>&1 > /dev/null
+	return $?
+}
+
+function install_git() {
+	echo "macOS developer tools are currently not installed which are required for installation skm."
+	read -p "Would you like to install them now? " -n 1 -r
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]
+	then
+		xcode-select --install
+	else
+    	echo "macOS developer tools not installed, please run: \"xcode-select --install\" in the terminal and then reinstall."
+		exit 1
+	fi
+}
+
+echo "Checking for macOS developer tools..."
+has_git
+if [ $? -ne 0 ]; then
+	install_git
+	has_git
+	if [ $? -ne 0 ]; then
+		printf "\t$MARK_CROSS Failed to install macOS developer tools. Please ensure you have \"git\" installed and try again.\n"
+		exit 1
+	fi
+else
+	printf "\t$MARK_CHECK macOS developer tools appears to be installed\n"
+fi
+
+echo "Checking for an existing version of SplashKit..."
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CROSS SplashKit is already installed!\n"
+	read -p "Would you like to completely re-install SplashKit? " -n 1 -r
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]; then
+		D=`date +%y%m%d-%H%M%S`
+		OLD_PATH="$INSTALL_PATH-$D"
+		echo "Removing existing SplashKit installation..."
+		mv $INSTALL_PATH $OLD_PATH
+		if [ -d "${INSTALL_PATH}" ]; then
+			printf "\t$MARK_CROSS failed to remove existing SplashKit installation.\n"
+			exit 1
+		else
+			printf "\t$MARK_CHECK removed existing SplashKit installation.\n"
+		fi
+	else
+		printf "\t$MARK_CHECK SplashKit is already installed.\n"
+		exit 1
+	fi
+fi
+
+echo "Installing SplashKit..."
+git clone --depth 1 --branch master $GIT_SKM_REPO "${INSTALL_PATH}"
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CHECK SplashKit was installed.\n"
+else
+	printf "\t$MARK_CROSS failed to install SplashKit.\n"
+	exit 1
+fi
+
+# Add to .bashrc if using bash and path line is missing for SplashKit inclusion.
+if [ ${SHELL} = "/bin/bash" ]; then
+	echo "Adding \"$INSTALL_PATH\" to your bash profile..."
+	grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.bash_profile || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.bash_profile
+    grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.bashrc || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.bashrc
+fi
+
+# Add to .zshrc if using zsh and path line is missing for SplashKit inclusion.
+if [ ${SHELL} = "/bin/zsh" ]; then
+	echo "Adding \"$INSTALL_PATH\" to your ZSH profile..."
+    grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.zshrc || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.zshrc
+fi
+
+printf "\t$MARK_CHECK done\n"
+export PATH="$INSTALL_PATH:$PATH"
+
+# Verify installation and access
+SKM_PATH="${INSTALL_PATH}/skm"
+if [ -f $SKM_PATH ]; then
+	printf "$MARK_CHECK SplashKit installed successfully!\n"
+else
+	printf "$MARK_CROSS SplashKit installation failed!\n"
+fi
+
+find "${INSTALL_PATH}" -name "*.sh" -exec chmod a+x "{}" \;
+
+which skm 2>&1 > /dev/null
+if [ $? -ne 0 ]; then
```

Comment:
**suggestion:** We can also collapse this to one line and fix up the [redirection order](https://github.com/koalaman/shellcheck/wiki/SC2069).

```suggestion
if which skm > /dev/null 2>&1; then
```